modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalEntries.scala

```diff
@@ -0,0 +1,93 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.syntax.all._
+
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.cats.eqv
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineV
+import fs2.data.csv.{CellDecoder, DecoderError}
+
+@derive(eqv)
+case class HistoricalSnapshotOrdinalEntries(values: List[HistoricalSnapshotOrdinalEntry])
+
+object HistoricalSnapshotOrdinalEntries {
+
+  implicit val historicalSnapshotOrdinalEntriesCellDecoder: CellDecoder[Option[HistoricalSnapshotOrdinalEntries]] =
+    CellDecoder.stringDecoder.emap { s =>
+      val timeline = mkSnapshotOrdinalTimeline(s)
+
+      if (timeline.isDefined)
+        timeline.asRight
+      else
+        new DecoderError("Unable to parse the ordinal timeline").asLeft
```

Comment:
It is an imperative conversion from `Option[A]` to `Either[DecoderError, A]`. You can write it declaratively by using `toRight` operator which just forces you to provide the "left side" (which is needed for Either but is missing in Option):

```scala
      mkSnapshotOrdinalTimeline(s)
        .toRight(new DecoderError("Unable to parse the ordinal timeline"))
        .map(_.some)
```